Model/Quote/SetQuoteAddresses.php

```diff
@@ -83,6 +83,8 @@ public function setAddresses(
         if ($billingAddress === null) {
             $quote->removeAddress($quote->getBillingAddress()->getId());
             $quote->removeAddress($quote->getShippingAddress()->getId());
+            $shippingAssignment = $this->shippingAssignmentProcessor->create($quote);
+            $quote->getExtensionAttributes()->setShippingAssignments([$shippingAssignment]);
```

Comment:
Do you need to create a shipping assignment? Could you not just do

```php
$quote->getExtensionAttributes()->setShippingAssignments([]);
```

Reply:
You're right, changed.